packages/pigeon/lib/cpp_generator.dart

```diff
@@ -901,6 +960,37 @@ EncodableValue ${api.name}::WrapError(const FlutterError& error) {
     indent.newln();
   }
 
+  void _writeClassConstructor(
+      Root root, Indent indent, Class klass, Iterable<NamedType> params) {
+    final Iterable<_HostNamedType> hostParams = params.map((NamedType param) {
+      return _HostNamedType(
+          _makeVariableName(param),
+          getFieldHostDatatype(param, root.classes, root.enums,
+              _shortBaseCppTypeForBuiltinDartType),
+          param.type);
+    });
+
+    String paramString = hostParams
+        .map((_HostNamedType param) =>
+            '\t${_hostApiArgumentType(param.hostType)} ${param.name}')
+        .join(',\n');
+    if (paramString.isNotEmpty) {
+      paramString = '\n$paramString\n';
+    }
+
+    String initializerString = hostParams
+        .map((_HostNamedType param) =>
+            '${param.name}_(${_fieldValueExpression(param.hostType, param.name)})')
+        .join(',\n\t\t');
+    if (initializerString.isNotEmpty) {
+      initializerString = ' : $initializerString';
+    }
+
+    indent.format('''
+${klass.name}::${klass.name}($paramString)$initializerString {}
+''');
```

Comment:
Seems like this could be done with a non-format method that would be easier to maintain.